src/python/pants/backend/python/util_rules/interpreter_constraints.py

```diff
@@ -513,3 +516,19 @@ def _major_minor_version_when_single_and_entire(ics: InterpreterConstraints) ->
 
     # anything else we don't understand
     raise _NonSimpleMajorMinor()
+
+
+def warn_on_python2_usage_in_interpreter_constraints(
+    interpreter_constraints: Iterable[str], *, description_of_origin: str
+) -> None:
+    @memoized
+    def inner(interpreter_constraints: tuple[str, ...]) -> None:
```

Comment:
I think the `memoized` here won't do quite the right thing, as each call of `warn_on_python2_usage_in_interpreter_constraints` will create a new `inner` and thus a new batch of memoization.

I think the `@memoized` needs to be a on a top level function?

Reply:
You're right. The `list`s being passed in to function in an earlier iteration are not hashable, so my brain was trying an "easy" fix to consolidate turning them into tuples for `@memoized` to work. Will just distribute the tuple calls to call sites like I should have done in the first place.